### ISSUE TITLE
Allow camel case on Message Binder special values

### ISSUE THREAD

**Issue by bjoernfi** [Caliburn-Micro/Caliburn.Micro#292]
While developing a windows 10 app I experienced the following issue when adding special values to the `MessageBinder` class via `MessageBinder.SpecialValues.Add`

If I select a variable name in camel case and use it in my view to inject it into the view model method, the **value is always null**. When changing the name to another one not involving camel case, everything works as expected. 

Code to reproduce:

```
MessageBinder.SpecialValues.Add("$chosenItem", (context) =>
{
    var eventArgs = context.EventArgs as AutoSuggestBoxSuggestionChosenEventArgs;
    return eventArgs?.SelectedItem;
});
```

```
<AutoSuggestBox x:Uid="ShellSearchBox" ItemsSource="{Binding NewsSearchResults}"
                            Margin="5" QueryIcon="Find" Width="250" TextMemberPath="Text"
                            PlaceholderText="Suche nach Nachrichten..." Grid.Column="2"
                            micro:Message.Attach="[Event QuerySubmitted] = [Action SearchNews($this.Text)];[Event TextChanged] = [Action SearchNews($this.Text)];[Event SuggestionChosen] = [Action NewsSearchResultChosen($chosenItem)]"/>
```

Change `$chosenItem` to `$chosenitem` in code above to resolve this issue. If this is not intended behaviour, I would offer my help to fix it with a pull request.

**Comment by nigel-sampson** [comment]
That does seem like an oversight, happy to take a pull request to solve this. I'd look at `MessageBinder.EvaluateParameter` as a good place.

**Comment by bjoernfi** [comment]
There you go, I also included a unit test for verifying that this bug has been solved.

**Comment by nigel-sampson** [comment]
Resolved by #294
